app/javascript/components/subject/MonitoringStatus.js

```diff
@@ -288,7 +288,22 @@ class MonitoringStatus extends React.Component {
           apply_to_group_cm_only_date: this.state.apply_to_group_cm_only_date,
         })
         .then(() => {
-          location.reload(true);
+          if (diffState.includes('jurisdiction_path')) {
+            const currentUserJurisdictionString = this.props.current_user.jurisdiction_path.join(', ');
+            // check if current_user has access to the changed jurisdiction
+            // if so, reload the page, if not, redirect to exposure or isolation dashboard
+            if (!this.state.jurisdiction_path.startsWith(currentUserJurisdictionString)) {
+              if (this.state.isolation) {
+                location.assign((window.BASE_PATH ? window.BASE_PATH : '') + '/public_health/isolation');
```

Comment:
Duplicate code here you can determine what the end of the path should be based on this.state.isolation and store that in a variable with a ternary. That way you don't have to write out `location.assign((window.BASE_PATH ? window.BASE_PATH : '')...` twice.

Reply:
all set and rebased